interview-questions.md

```diff
@@ -8,50 +8,104 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Iteration vs recursion. Any recursive solution can be turned into an iterative one. My understanding is that iterative is often faster and saves on memory when compared to the recursive method. However the recursive method is often much easier to write and to read and presents more elegant code.
 
 5. What is the difference between a class and an object?
 
-Your answer:
+Your answer: A class is the blueprint for an object. An object is a data structure that holds information and functions called methods. 
 
-Researched answer:
+Researched answer: So class is a blueprint for an object while an object is an instance of a class. No memory is allocated when a class is created but objects do take up memory when created. The class is declared only once while an object can be created as many times as you want from the class blueprint. A class can not be manipulated since it isn't in memory but objects can be manipulated using dot notation. A class is a logical entity while an object is a physical entity. Not quite sure what that means. A class is declared with the class keyword while the object is declared with the new keyword. Important take away is a class is not an object but just the blueprint for creating objects.
 
 6. STRETCH: What is hoisting in JavaScript?
 
-Your answer:
+Your answer: I am not very sure on this but I believe hoisting is about scope and where something lives in the order of operations for running the code. I do not think javascript is purely run top to bottom but it is read through first and certain elements in javascript will be hoisted to the top of the order so that they can be accessed immediately no matter where they live in the actual document. I will do some research and get back to you on the specifics.
+
+Researched answer: Javascript hoisting is where a variable or function is available before it's decleration. The only thing that can do this is function and async function declerations. Var can also do this to a lesser extent.
+function myfunc(){
+    return something
+}
+This function or it's async varient will be hoisted and be available to use even if it is used before it is defined. So:
+myFunc()
+function myfunc(){
+    return something
+}
+should work. Also:
+x = 10
+var x
+will also work even though var x is defined after.
+
+let and const are also hoisted to the top of their respective block but let will throw a ReferenceError if you try to use it before its decleration and const will just not run at all. 
 
-Researched answer:
+In short declerations are hoisted while initializion is not. Thefore declare your variables at the top of each scope it is used in.
```

Comment:
Amazing work in this section too Garrett!!  You did a great job in really digging deep into the concepts both in your answers and explored those concepts even further with your research.  Exactly what I am looking for 🎉